### PR TITLE
Refactor pipeline to run tests in public Semaphore

### LINKED ISSUE #234: CI jobs don't run for external pull requests
Example: https://github.com/confluentinc/vscode/pull/219

When pull request coming from a forked repo (i.e. external contribution) CI jobs to run linters and tests do not start.

### PR DESCRIPTION
**Pull request by rohitsanj**
## Summary of Changes

- Skip vault setup in public Semaphore
- Fetch sidecar executable using cURL in public Semaphore
- See #235 for passing public Semaphore pipeline with the same changeset as this PR. Successfully runs lint + tests.

Resolves #234

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,19 +100,21 @@ download-sidecar-executable:
...
+			chmod +x $(EXECUTABLE_DOWNLOAD_PATH) && \
+			echo "Downloaded sidecar executable to $(EXECUTABLE_DOWNLOAD_PATH)"; \
+	else \
+		mkdir -p bin && \
+		echo "Using gh to download sidecar executable from GitHub release $(IDE_SIDECAR_VERSION)"; \
+		export EXECUTABLE_PATH=ide-sidecar-$(IDE_SIDECAR_VERSION_NO_V)-runner-$(SIDECAR_OS_ARCH) && \
+			gh release download $(IDE_SIDECAR_VERSION) --repo $(IDE_SIDECAR_REPO) --pattern=$${EXECUTABLE_PATH} --output $(EXECUTABLE_DOWNLOAD_PATH) --clobber && \
+			chmod +x $(EXECUTABLE_DOWNLOAD_PATH) && \
+			echo "Downloaded sidecar executable to $(EXECUTABLE_DOWNLOAD_PATH)"; \
+	fi
```
Comment (rhauch) [inline Makefile:117]: I get that you just changed the indentation of these lines and modified the condition on line 103, but why do we need to use two different techniques to download from the ide-sidecar releases?

Can we just use one of these approaches?

**Review by rhauch — COMMENTED** [review]
Thanks, @rohitsanj. Great work, and nice simple improvement to enable public PR builds. 

However, I do have one question about using two different tools to download ide-sidecar releases. Do we need them both, and if so why? But since this isn't strictly speaking related to the subject of this PR, I'm fine with making any changes (if needed) in future PRs, if that's what you prefer.

File: Makefile (hunk 1 above)
Comment (rohitsanj) [reply Makefile:117]: My rationale was that if we could use the authenticated `gh` CLI to issue the download request, then we should do so, in order to leverage a higher tier of GitHub API performance and rate limiting. (I'm not sure if this is actually the case).

Having said that, I would be fine with simplifying down to just the `curl` approach. The change can be made in a follow-up PR, like you suggested. Thanks!

Comment (rhauch) [reply Makefile:117]: Yeah, might be worth it to simplify the logic in a future PR. Thanks!

**Review by rhauch — APPROVED** [review]

**Pull request merged**